sotodlib/site_pipeline/make_nearby_sources_db.py

```diff
@@ -0,0 +1,254 @@
+        max_ctime: Optional[float] = None,
+        obs_id: Optional[str] = None,
+        update_delay: Optional[str] = None,
+        ):
+    """
+    Args:
+        config: str
+            Path to the configuration file containing processing parameters.
+        output_dir: str or None
+            Directory where the processed data will be saved. \
+            If not provided, it is extracted from the configuration file.
+        verbose: str
+            Verbosity level for logging. Default is 2 (INFO).
+        overwrite: bool
+            Whether to overwrite existing output files. Default is False.
+        query_text: str or None
+            Text-based query to filter observations. If not provided, it is extracted from the configuration file.
+        query_tags: list or None
+            Tags used for further filtering observations. If not provided, it is extracted from the configuration file.
+        min_ctime: float or None
+            Minimum timestamp of observations to be queried. If not provided, it is extracted from the configuration file.
+        max_ctime: float or None
+            Maximum timestamp of observations to be queried. If not provided, it is extracted from the configuration file.
+        obs_id: str or None
+            Specific observation obs_id to process. If provided, overrides other filtering parameters.
+        update_delay: str or None
+            Number of days (unit is days) in the past to start observation list.
+    """
+    # Set verbose
+    if verbose == 0:
+        logger.setLevel(logging.ERROR)
+    elif verbose == 1:
+        logger.setLevel(logging.WARNING)
+    elif verbose == 2:
+        logger.setLevel(logging.INFO)
+    elif verbose == 3:
+        logger.setLevel(logging.DEBUG)
+    
+    # load configuration file
+    if type(config) == str:
+        with open(config) as f:
+            config = yaml.safe_load(f)
+    
+    # load context
+    context_file = config['context_file']
+    ctx = core.Context(context_file)
+    
+    if output_dir is None:
+        if 'output_dir' in config.keys():
+            output_dir = config['output_dir']
+        else:
+            raise ValueError('output_dir is not specified.')
+    if query_text is None:
+        if 'query_text' in config.keys():
+            query_text = config['query_text']
+    if query_tags is None:
+        if 'query_tags' in config.keys():
+            query_tags = config['query_tags']
+    if min_ctime is None:
+        if 'min_ctime' in config.keys():
+            min_ctime = config['min_ctime']
+    if max_ctime is None:
+        if 'max_ctime' in config.keys():
+            max_ctime = config['max_ctime']
+    if update_delay is None:
+        if 'update_delay' in config.keys():
+            update_delay = config['update_delay']
+    
+    if (min_ctime is None) and (update_delay is not None):
+        min_ctime = int(time.time()) - update_delay*86400
+        logger.info(f'min_ctime: {min_ctime}')
```

Comment:
You should probably raise an error if update_delay is specified, but then ignored (because min_ctime was specified in the config file or the command line).  I would expect "update_delay" to let me speed things up by only looking a little bit into the past.  But this doesn't override an existing min_ctime... is that the behavior you want?